drivers/sub8_videoray_m5_thruster/nodes/thruster_driver.py

```diff
@@ -14,9 +14,58 @@
 from ros_alarms import AlarmBroadcaster, AlarmListener
 lock = threading.Lock()
 
+class BusVoltageEstimator(object):
+    '''
+    Class that estimates sub8's thruster bus voltage.
+    As of May 2017, this is just a simple rolling average with a constant width sliding
+    window. However add_reading and get_estimate methods are left for when smarter
+    filtering is needed
+    '''
+    class VoltageReading(object):
+        def __init__(self, voltage, time):
+            self.v = voltage
+            self.t = time
+
+    def __init__(self, window_duration):
+         '''
+         window_duration - float (amount of seconds for which to keep a reading in the buffer)
+         '''
+         self.window_duration = rospy.Duration(window_duration)
+         self.last_update_time = None
+         self.buffer = []
+
+    def add_reading(self, voltage, time):
+        ''' Adds voltage readings to buffer '''
+        self.buffer.append(self.VoltageReading(voltage, time))
+        self.last_update_time = time
+        self.prune_buffer()
+
+    def prune_buffer(self):
+        ''' Removes readings older than the window_duration from buffer '''
+        for reading in self.buffer:
+            age = rospy.Time.now() - reading.t
+            if age > self.window_duration:
+                self.buffer.remove(reading)
+
+    def get_voltage_estimate(self):
+        ''' Returns average voltage in buffer '''
+        voltages = []
+        for r in self.buffer:
+            voltages.append(r.v)
+        if not self.buffer:
+            return None
+        else:
+            return np.mean(voltages)
+
+    def get_last_update_time(self):
+        ''' Returns time of most recent reading in buffer '''
+        if self.last_update_time is None:
+            self.last_update_time = rospy.Time.now()
+        return self.last_update_time
 
 class ThrusterDriver(object):
-    _dropped_timeout = 1 # s
+    _dropped_timeout = 1.0 # s
+    _window_duration = 30.0 # s
```

Comment:
I disagree. I can see it killing after a 10 seconds where we would have come right back above the kill voltage at the end of some move.

However, running it for 20 extra seconds after we actually get to a killable (steady state) voltage does not make much of a difference.